### PR TITLE
[FIX] all: remove en_US as fallback for lang code

### PR DESCRIPTION
**Pull request by william-andre**
We are not sure en_US is installed even if it is the default language
because it is not installed if the server was installed with
--load_language (different from en_US)

This commit was done because when trying to install account, the tax
return entry tried to format a date at the installation of the module,
with no lang in the context.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/models.py [hunk 1]
```diff
@@ -1851,7 +1851,7 @@ def _read_group_fill_temporal(self, data, groupby, aggregated_fields, annotated_
     def _read_group_prepare(self, orderby, aggregated_fields, annotated_groupbys, query):
         """
         Prepares the GROUP BY and ORDER BY terms for the read_group method. Adds the missing JOIN clause
-        to the query if order should be computed against m2o field. 
+        to the query if order should be computed against m2o field.
```
Comment (mart-e) [inline odoo/models.py]: It seems you have an automatic tool/linter that remove trailing spaces. While it is a good for new developments, we avoid changing it in existing files, it makes a bigger diff and creates potential conflicts.

File: addons/account/models/account_journal_dashboard.py [hunk 2]
```diff
@@ -281,11 +281,6 @@ def get_journal_dashboard_datas(self):
             if read:
                 number_to_check = read[0]['__count']
                 to_check_balance = read[0]['amount_total']
-        elif self.type == 'general':
-            read = self.env['account.move'].read_group([('journal_id', '=', self.id), ('to_check', '=', True)], ['amount_total'], 'journal_id', lazy=False)
```
Comment (mart-e) [inline addons/account/models/account_journal_dashboard.py]: Is it expected that this part is in this PR? Looks unrelated

Comment (william-andre) [reply addons/account/models/account_journal_dashboard.py]: Indeed, bad rebase I guess

File: odoo/tools/misc.py [hunk 3]
```diff
@@ -1162,6 +1162,14 @@ def html_escape(text):
     def html_escape(text):
         return werkzeug.utils.escape(text)
 
+
+def get_lang(env, lang_code=False):
```
[the remark this answers is not in the document]
Comment (william-andre) [reply odoo/tools/misc.py]: I tried to do something like you wanted but I still think this is a better solution. I changed it to be more robust though.

File: odoo/tools/misc.py [hunk 4]
```diff
@@ -1221,7 +1227,7 @@ def format_date(env, value, lang_code=False, date_format=False):
         else:
             value = odoo.fields.Datetime.from_string(value)
 
-    lang = env['res.lang']._lang_get(lang_code or env.context.get('lang') or 'en_US')
+    lang = get_lang(env, lang_code)
```
[the remark this answers is not in the document]
Comment (william-andre) [reply odoo/tools/misc.py]: But then I would have to do it at lines 1250, 1281, 1311 and 1329, wouldn't I? This is the kind of things that made one method work but not the others because a fix is rarely done everywhere (like it should) but only at one place.

File: odoo/tools/misc.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (william-andre) [reply odoo/tools/misc.py]: And I added the docstring

**Comment by william-andre** [comment]
@mart-e This is a bit out of the scope of this commit, but we could think of putting this in `tools/misc.py` too?
https://github.com/odoo/odoo/blob/cef2bbd12f9982a59bab3c3066afc8e48e55c023/addons/http_routing/models/ir_http.py#L337-L351

This way, everything related to lang could be done in one single function.

**Comment by mart-e** [comment]
@william-andre this `get_nearest_lang` is strongly linked to website, see #37494 for the kind of issues we have with it.
I agree it would makes sense to move it in a refactoring but I am not convinced of the purpose of this method in the first place. Trying to find a lang that +/- looks like another leads to issues like using Croatian code (`hr_HR`) to match Human Resources path (`hr_org` module) ^^

**Comment by mart-e** [comment]
Ok for the more robust change, thanks for the update.
I removed the `env.user.lang` in the list as the previous code in `formatLang` used the company.

@robodoo if this is good for @qdp-odoo too you can r+
